PowerValidatedSolutions.psm1

```diff
@@ -12939,6 +12939,127 @@ Function Test-vROPsAdapterStatusByType {
 }
 Export-ModuleMember -Function Test-vROPsAdapterStatusByType
 
+Function Update-vROPSvRAAdapterCredential {
+    <#
+		.SYNOPSIS
+        Update the credential of vRealize Automation adapter in vRealize Operations Manager
+
+        .DESCRIPTION
+        The Update-vROPSvRAAdapterCredential cmdlet update the credential of vRealize Automation adapter in vRealize Operations Manager. The
+        cmdlet connects to SDDC Manager using the -server, -user, and -password values.
+        - Validates that network connectivity and authentication is possible to SDDC Manager
+        - Validates that vRealize Operations Manager has been deployed in VMware Cloud Foundation aware mode and retrieves its details
+        - Validates that network connectivity and authentication is possible to vRealize Operations Manager
+        - Validates that vRealize Automation has been deployed in VMware Cloud Foundation aware mode and retrives its details
+        - Validates that network connectivity is possible to vRealize Automation
+        - Validates that the vRealize Automation adapter exists and configured in vRealize Operations Manager
+        - Validates that the credential name of vRealize Automation adapter does not already exist in vRealize Operations Manager
+        - Validates that the given credential is valid and updates vRealize Automation adapter in vRealize Operations Manager 
+        - Verifies the vRealize Automation adapter status in vRealize Operations Manager after updating the credential
+
+        .EXAMPLE
+        Update-vROPSvRAAdapterCredential -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -vRACredName "vRealize Automation Credentials" -vRAcredUserName svc-vrops-vra@sfo.rainpole.io -vRACredPass VMw@re1! 
+        This example update the credential of vRealize Automation adapter with name "vRealize Automation Credentials" in vRealize Operations Manager
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$vRACredName,
```

Comment:
Would it be more appropriate to use simplified parameters like:

- `credential_name` (or `credential_displayname`)
- `credential_username`
- `credential_password`